### ISSUE TITLE
Subscriber never receiving message Fan out

### ISSUE THREAD

**Issue by artandor** [erlio/vmq_mzbench#28]
Hi, I'm currently benchmarking mqtt for both rabbitmq, vernemq and activemq.

I think mzbench is the most valuable tool to do this, but I struggle pretty hard. My subsciribers never receive the produced messages. I tried a lot of custom scenario, but even the official scenarios from this repo doesn't work.

Here is the scenario.

```
#!benchDL
#######
# Broadcast Scenario:
# 500 subscribers reading from the same topic "fixed/broadcast/topic"
# 1 publisher sending 1 msg/s to topic "fixed/broadcast/topic"
# Overall Msg rate: 500 msg/s
# Message Size: 150 random bytes
# Runtime: 5 min
#######

make_install(git = "https://github.com/erlio/vmq_mzbench.git",
             branch = "master")
             
defaults("pool_size" = 500)

pool(size = numvar("pool_size"),
     worker_type = mqtt_worker,
     woker_start = poisson(1000 rps)):

            connect(host = "some-ip",
                    port = 1883,
                    client = fixed_client_id("subscriber_pool1", worker_id()),
                    clean_session = true,
                    keepalive_interval = 60,
                    proto_version = 4,
                    reconnect_timeout = 4
                    )

            set_signal(subscribe1, 1)
            wait_signal(subscribe1, numvar("pool_size"))
            wait(10 sec)
            subscribe("fixed/broadcast/topic", 0)


pool(size = 1,
     worker_type = mqtt_worker):

            connect(host = "some-ip",
                    port = 1883,
                    client = fixed_client_id("publisher_pool1", worker_id()),
                    clean_session = true,
                    keepalive_interval = 60,
                    proto_version = 4,
                    reconnect_timeout = 4
                    )

            set_signal(connect1, 1)
            wait_signal(connect1, 1)
            wait(4 sec)
            loop(time = 5 min, rate = 1 rps):
                publish("fixed/broadcast/topic", random_binary(150), 0)
            disconnect()
```

Thanks in advance for answering, this is a pretty urgent benchmark :)

Have a nice day

**Comment by ioolkos** [comment]
Hi @artandor thanks for the issue. 
Can you tell me what the MZbench WebGUI tells you about the error you see?

In other words: does the test script start running, or not? do you see latency graph for the QoS 0 metrics?

**Comment by artandor** [comment]
The job will run, publishing will work well, but subscribing won't do anything.

I took screenshots of the graphs.

Here is the screen for the mqtt internal stats : 
![image](https://user-images.githubusercontent.com/22364556/49872358-1719bb80-fe19-11e8-8d5d-edd80e88a973.png)


**And here are the screens for most importants boards in the 1 to 1 scenario.**

![image](https://user-images.githubusercontent.com/22364556/49872171-9490fc00-fe18-11e8-97fa-9e98c3438bbf.png)


![image](https://user-images.githubusercontent.com/22364556/49872208-a7a3cc00-fe18-11e8-81c9-3ff747117115.png)


![image](https://user-images.githubusercontent.com/22364556/49872236-c4400400-fe18-11e8-93d9-51f57a9e5c1e.png)

Thanks a lot for your fast answer :)

**Comment by ioolkos** [comment]
In the "MQTT connections pane" do you have 500 or 501 connections? (1 publisher).

**Comment by ioolkos** [comment]
Oh, and start playing around with the timing. Slow down the connection setup, for instance, instead of poisson 1000, set it to poisson 100. Then wait with the publisher for a little longer before you start publishing: wait(20 sec) or something. The underlying bench client is brutally asynchrounous which leads to strange timing effects possibly.

EDIT: on my laptop your script works as is.

**Comment by artandor** [comment]
What broker are you benching on ?

I had 501 connections
![image](https://user-images.githubusercontent.com/22364556/49873746-ee93c080-fe1c-11e8-8c12-83baf2bedf09.png)

So you recommend to set 20s wait before starting the production loop ?

**Comment by ioolkos** [comment]
I use this to bench VerneMQ. (what broker are you testing?)

**Comment by artandor** [comment]
I'm testing it to bench on rabbitmq and activemq at the moment

Edit0 : Ok so, it's working on activemq now, but not on rabbitmq. Is there vernemq specific code in your package ? 

Edit1: I'm still trying to find out what's wrong with RabbitMQ specificaly, I'll post an update and close the ticket once i find out.

**Comment by larshesel** [comment]
Hi @artandor did you figure out what the problem was?

**Comment by artandor** [comment]
Hi, no i did not.
